app/scripts/controllers/preferences.js

```diff
@@ -767,189 +610,4 @@ export default class PreferencesController {
 
     this.store.updateState({ infuraBlocked: isBlocked });
   }
-
-  /**
-   * Updates `accountTokens`, `tokens`, `accountHiddenTokens` and `hiddenTokens` of current account and network according to it.
-   *
-   * @param {array} tokens - Array of tokens to be updated.
-   * @param {array} assetImages - Array of assets objects related to assets added
-   * @param {array} hiddenTokens - Array of tokens hidden by user
-   *
-   */
-  _updateAccountTokens(tokens, assetImages, hiddenTokens) {
-    const {
-      accountTokens,
-      chainId,
-      selectedAddress,
-      accountHiddenTokens,
-    } = this._getTokenRelatedStates();
-    accountTokens[selectedAddress][chainId] = tokens;
-    accountHiddenTokens[selectedAddress][chainId] = hiddenTokens;
-    this.store.updateState({
-      accountTokens,
-      tokens,
-      assetImages,
-      accountHiddenTokens,
-      hiddenTokens,
-    });
-  }
-
-  /**
-   * Detects whether or not a token is ERC-721 compatible.
-   *
-   * @param {string} tokensAddress - the token contract address.
-   *
-   */
-  async _detectIsERC721(tokenAddress) {
```

Comment:
So this detection is now handled in the assets controller?

Reply:
in the TokensController yes: https://github.com/MetaMask/controllers/blob/176418fc9411fba27edcdb1b92f8a857a8e8700c/src/assets/TokensController.ts#L326